src/impls/primitive.rs

```diff
@@ -90,7 +89,41 @@ macro_rules! ImplDekuRead {
                         <$typ>::from_be_bytes(bytes.try_into()?)
                     }
                 };
+                Ok((rest, value))
+            }
+        }
 
+        impl DekuRead<'_, (Endian, ByteSize)> for $typ {
+            fn read(
+                input: &BitSlice<Msb0, u8>,
+                (endian, size): (Endian, ByteSize),
+            ) -> Result<(&BitSlice<Msb0, u8>, Self), DekuError> {
+                let max_type_bits: usize = BitSize::of::<$typ>().0;
+                let bit_size: usize = size.0 * 8;
+
+                let input_is_le = endian.is_le();
+
+                if bit_size > max_type_bits {
+                    return Err(DekuError::Parse(format!(
+                        "too much data: container of {} bits cannot hold {} bits",
+                        max_type_bits, bit_size
+                    )));
+                }
+
+                if input.len() < bit_size {
+                    return Err(DekuError::Incomplete(crate::error::NeedSize::new(bit_size)));
+                }
+
+                let (bit_slice, rest) = input.split_at(bit_size);
+
+                let bytes: &[u8] = bit_slice.as_raw_slice();
```

Comment:
This is exactly what I want in that comment. But... 
> the removal of this branch for bytes
> ```rust
> if pad == 0
> && bit_slice.len() == max_type_bits
> && bit_slice.as_raw_slice().len() * 8 == max_type_bits
> ```
Can we really do this? What if `bit_slice` is not byte aligned?

Reply:
We should be good given:
- `let bit_size: usize = size.0 * 8;` -- will always be byte aligned
- we check `bit_size > max_type_bits` & `input.len() < bit_size` before calling `split_at`